### PR TITLE
Piyush whitespace issue

### PR DESCRIPTION
**Pull request by phadej**
Rebased to current (2019-03-02) `master`

Unconflicts and improves over https://github.com/haskell/cabal/pull/5293

### CODE REVIEW

File: Cabal/Distribution/Types/ModuleRenaming.hs [hunk 1]
```diff
@@ -67,6 +68,19 @@ isDefaultRenaming :: ModuleRenaming -> Bool
 isDefaultRenaming DefaultRenaming = True
 isDefaultRenaming _ = False
 
+-- | For versions < 3.0 white spaces were not skipped after the '('
```
Comment (hvr) [inline Cabal/Distribution/Types/ModuleRenaming.hs]: this comment needs updating?

File: Cabal/tests/ParserTests/regressions/mixin-2.cabal [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+version: 0
+
+-- mixin field:
+-- in 2.2 we got leading/trailing commas
+-- in 2.2 we got lax space parsing
+executable str-example
+  main-is:             Main.hs
+  build-depends:       base, str-string, str-bytestring
+  mixins:              str-string     ( Str as Str.String     ),
+                       str-bytestring ( Str as Str.ByteString ),
```
Comment (quasicomputational) [inline Cabal/tests/ParserTests/regressions/mixin-2.cabal]: Can we test the `hiding` variant, as well? That's also changed.

**Comment by phadej** [comment]
I just resolved the conflict of previous PR, and added tests which should been there. If someone really want this for 2.4, they can resolve the remaining issues.

File: Cabal/tests/ParserTests/regressions/mixin-2.cabal (hunk 2 above)
Comment (phadej) [reply Cabal/tests/ParserTests/regressions/mixin-2.cabal]: Feel free to add.

Comment (phadej) [reply Cabal/tests/ParserTests/regressions/mixin-2.cabal]: In fact, yes, we should add the test. But I don't have the bandwidth to do it.

**Comment by piyush-kurur** [comment]
@phadej what are the missing things now? I can try to see if I can contribute so that we have this in 2.4

**Comment by quasicomputational** [comment]
@piyush-kurur There was a failure on Travis but I don't see it now, so maybe that's been fixed. There are only two things that I think need doing now:

* analogous tests for the `hiding` form, which have also changed (i.e., make sure whitespace breaks the 2.2 parser but not the 2.4 one)
* and docs + changelogs

**Comment by phadej** [comment]
Ah, I should fix this for 3.0

**Comment by phadej** [comment]
Rebased on current master, let's get this in for 3.0

**Comment by piyush-kurur** [comment]
Aha finally my issues with white space has been cured ;-)

**Pull request merged**